### PR TITLE
Dicom update API - Added DICOM100 prefix for tag and value validation errors

### PR DESCRIPTION
**Pull request by arunmk-ms**
## Description
Added DICOM100 prefix for tag and value validation errors for Dicom update API.
Updated unit tests 

## Testing
Ran unit tests and verified the API manually.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs [hunk 1]
```diff
@@ -52,15 +55,20 @@ public class UpdateOperationInstanceService : IUpdateOperationInstanceService
...
         EnsureArg.IsNotNull(updateSpecification, nameof(updateSpecification));
         EnsureArg.IsNotNull(updateSpecification.ChangeDataset, nameof(updateSpecification.ChangeDataset));
 
         UpdateRequestValidator.ValidateRequest(updateSpecification);
-        UpdateRequestValidator.ValidateDicomDataset(updateSpecification.ChangeDataset);
+        DicomDataset failedSop = UpdateRequestValidator.ValidateDicomDataset(updateSpecification.ChangeDataset);
+
+        if (failedSop.Any())
+        {
+            return new UpdateInstanceResponse(failedSop, (int)HttpStatusCode.BadRequest);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs:70]: Should the core needs to know the status. Can we infer from the failure itself?

File: src/Microsoft.Health.Dicom.Core/Messages/Update/UpdateInstanceResponse.cs [hunk 2]
```diff
@@ -4,16 +4,28 @@
...
+    public UpdateInstanceResponse(OperationReference operationReference, int statusCode)
     {
         Operation = EnsureArg.IsNotNull(operationReference);
+        StatusCode = statusCode;
+    }
+
+    public UpdateInstanceResponse(DicomDataset dataset, int statusCode)
+    {
+        FailedDataset = EnsureArg.IsNotNull(dataset);
+        StatusCode = statusCode;
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Messages/Update/UpdateInstanceResponse.cs:23]: We can set the Statuscode based on the FailedDataset

File: src/Microsoft.Health.Dicom.Core.UnitTests/Features/Update/UpdateRequestValidatorTests.cs [hunk 3]
```diff
@@ -54,9 +54,11 @@ public void GivenAValidDataset_WhenValidated_ThenItShouldSucceed(DicomDataset da
 
     [Theory]
     [MemberData(nameof(GetInvalidDicomDataset))]
-    public void GivenAnInvalidDataset_WhenValidated_ThenExceptionShouldBeThrown(DicomDataset dataset)
+    public void GivenAnInvalidDataset_WhenValidated_ThenSucceedsWithErrorsInFailedAttributesSequence(DicomDataset dataset)
     {
-        Assert.Throws<BadRequestException>(() => UpdateRequestValidator.ValidateDicomDataset(dataset));
+        DicomDataset failedSop = UpdateRequestValidator.ValidateDicomDataset(dataset);
+        DicomSequence failedAttributeSequence = failedSop.GetSequence(DicomTag.FailedAttributesSequence);
+        Assert.Single(failedAttributeSequence);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core.UnitTests/Features/Update/UpdateRequestValidatorTests.cs:61]: Could we validate the message too?

File: src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs [hunk 4]
```diff
@@ -4,14 +4,17 @@
...
+using FellowOakDicom;
 using Microsoft.Health.Dicom.Core.Exceptions;
 using Microsoft.Health.Dicom.Core.Features.Common;
 using Microsoft.Health.Dicom.Core.Features.Context;
 using Microsoft.Health.Dicom.Core.Features.Operations;
 using Microsoft.Health.Dicom.Core.Features.Routing;
+using Microsoft.Health.Dicom.Core.Messages.Update;
 using Microsoft.Health.Dicom.Core.Models.Operations;
 using Microsoft.Health.Dicom.Core.Models.Update;
 using Microsoft.Health.Operations;
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs:20]: nit: I updated the class name to `UpdateInstanceOperationService`

File: src/Microsoft.Health.Dicom.Core.UnitTests/Features/Update/UpdateRequestValidatorTests.cs (hunk 3 above)
Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Core.UnitTests/Features/Update/UpdateRequestValidatorTests.cs:62]: Good point. I added the check for verifying error comment too.

File: src/Microsoft.Health.Dicom.Core/Features/Update/UpdateRequestValidator.cs [hunk 5]
```diff
@@ -48,43 +49,37 @@ public static void ValidateRequest(UpdateSpecification updateSpecification)
...
         {
-            throw new BadRequestException(errorString);
+            var failedAttributes = errors.Select(error => new DicomDataset(new DicomLongString(DicomTag.ErrorComment, error))).ToArray();
+            var failedAttributeSequence = new DicomSequence(DicomTag.FailedAttributesSequence, failedAttributes);
+            failedSop.Add(failedAttributeSequence);
         }
+        return failedSop;
     }
 
     private static string GetCommaSeparatedTags(List<string> tags)
```
[the remark this answers is not in the document]
Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Update/UpdateRequestValidator.cs:85]: Yes, that is how we return the error comment.

File: src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs (hunk 4 above)
Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Update/UpdateOperationInstanceService.cs:19]: I updated the class and interface accordingly.

**Review by bcarthic — APPROVED** [review]

**Pull request merged**